tests/test_boltzmann_machine.py

```diff
@@ -78,6 +88,43 @@ def test_forward(self):
             en_boltz = self.bm(fake_spins).item()
             self.assertAlmostEqual(en_bqm, en_boltz, 4)
 
+    def test_estimate_beta(self):
```

Comment:
This could use some parameterization (worth including [parameterized](https://pypi.org/project/parameterized/) as a dev dependency IMO).